src/coreclr/vm/threads.h

```diff
@@ -4509,6 +4509,17 @@ class Thread
     #endif
     }
 
+    static bool UseRedirectForGcStress()
+    {
+        LIMITED_METHOD_CONTRACT;
+
+#ifdef USE_REDIRECT_FOR_GCSTRESS
+        return UseContextBasedThreadRedirection();
```

Comment:
Can we just always return `false`?   (and delete all the support for GC stress redirection)

Reply:
Unfortunately not. I have originally just undefined the USE_REDIRECT_FOR_GCSTRESS and fixed few places, but I've hit a race problem between another thread redirection and the GC stress running from VEH updating GC references in the context passed to the VEH. I've added various stresslog loggings and found that there is a race when the thread was returning from the VEH and another thread redirected it at that point, the other thread got GetThreadContext with the old values of those GC references. And the context didn't have any of the indicators that would make us not to use it.
I was able to repro it only on the CET enabled machine that has a beta version of Windows, so it could have been caused by a bug that was introduced in that version of Windows. So I've decided to use it only when redirection is disabled (which is currently the case for CET only).